### PR TITLE
Correct some typos

### PR DESCRIPTION
**Pull request by bouweandela**
Spelling mistakes and typos found with [codespell](https://github.com/codespell-project/codespell) and [misspell](https://github.com/client9/misspell).

### CODE REVIEW

File: doc/esmvalcore/config.rst [hunk 1]
```diff
@@ -43,7 +43,7 @@ option:
   # turning it off will turn off graphical output from diagnostic
   write_plots: true
 
-  # Diagnositcs write NetCDF files? [true]/false
+  # Diagnostic write NetCDF files? [true]/false
```
Comment (mattiarighi) [inline doc/esmvalcore/config.rst]: But then: `write``--> `writes`.
Although I find the plural better.

File: doc/esmvalcore/config.rst [hunk 2]
```diff
@@ -113,7 +113,7 @@ Most of these settings are fairly self-explanatory, e.g.:
 
   # Diagnostics create plots? [true]/false
   write_plots: true
-  # Diagnositcs write NetCDF files? [true]/false
+  # Diagnostic write NetCDF files? [true]/false
```
Comment (mattiarighi) [inline doc/esmvalcore/config.rst]: Same here.

**Comment by mattiarighi** [comment]
Thanks for the spellcheck!

**Comment by mattiarighi** [comment]
Should we add the two above packages to our installation setup?

**Comment by bouweandela** [comment]
I'll make a pull request to add codespell to the developer dependencies

**Comment by bouweandela** [comment]
misspell is a go package, so not so easy to add

**Pull request merged**